metricflow/query/query_parser.py

```diff
@@ -529,6 +537,34 @@ def _validate_order_by_specs(
             ):
                 raise InvalidQueryException(f"Order by item {order_by_spec} not in the query")
 
+    def _validate_date_part(
+        self, metric_references: Sequence[MetricReference], time_dimension_specs: Sequence[TimeDimensionSpec]
+    ) -> None:
+        """Validate that date parts can be used for metrics.
+
+        TODO: figure out expected behavior for date part with these types of metrics.
+        """
+        date_part_requested = False
+        for time_dimension_spec in time_dimension_specs:
+            if time_dimension_spec.date_part:
+                date_part_requested = True
+                if time_dimension_spec.date_part.to_int() < time_dimension_spec.time_granularity.to_int():
+                    raise RequestTimeGranularityException(
+                        f"Date part {time_dimension_spec.date_part.name} is not compatible with time granularity "
+                        f"{time_dimension_spec.time_granularity.name}."
+                    )
+        if date_part_requested:
+            for metric_reference in metric_references:
+                metric = self._metric_lookup.get_metric(metric_reference)
+                if metric.type == MetricType.CUMULATIVE:
+                    raise UnableToSatisfyQueryError("Cannot extract date part for cumulative metrics.")
```

Comment:
Huh. I don't think we should do this here, because none of the other validate methods do it, but now I'm wondering if we should validate everything and raise one exception with all the problems, instead of this piecemeal thing.

Reply:
I put this here because it was similar to what's happening in `_validate_no_time_dimension_query()`. Where else would you expect this to live?
I don't disagree about raising one exception, but not sure it's in scope for this PR.